src/js/popup/popup.js

```diff
@@ -1,802 +1,1350 @@
+            serverStoragePref,
+            "server_storage"
+          );
+
+        if (!serverStoragePrefInLocalStorage) {
+          // There is no reference to the users storage preference saved. Fetch it from the server.
+          return await browser.runtime.sendMessage({
+            method: "getServerStoragePref",
+          });
+        } else {
+          // If the stored pref exists, return value
+          return serverStoragePref.server_storage;
+        }
+      },
+      getCurrentPage: async () => {
+        const [currentTab] = await browser.tabs.query({
+          active: true,
+          currentWindow: true,
+        });
+        return currentTab;
+      },
+      getMasks: async (options = { fetchCustomMasks: false }) => {
+        const serverStoragePref =
+          await popup.utilities.getCachedServerStoragePref();
+
+        if (serverStoragePref) {
+          try {
+            return await browser.runtime.sendMessage({
+              method: "getAliasesFromServer",
+              options,
+            });
+          } catch (error) {
+            console.warn(`getAliasesFromServer Error: ${error}`);
+
+            // API Error — Fallback to local storage
+            const { relayAddresses } = await browser.storage.local.get(
+              "relayAddresses"
+            );
+
+            return relayAddresses;
+          }
+        }
+
+        // User is not syncing with the server. Use local storage.
+        const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+        return relayAddresses;
+      },
+      populateNewsFeed: async ()=> {
+        // audience can be premium, free, phones, all
+        // Optional data: waffle, fullCta*
+        const savings = "40%"; // For "Save 40%!" in the Bundle promo body
+        
+        const isBundleAvailableInCountry = (await browser.storage.local.get("bundlePlans")).bundlePlans.BUNDLE_PLANS.available_in_country;
+        const isPhoneAvailableInCountry = (await browser.storage.local.get("phonePlans")).phonePlans.PHONE_PLANS.available_in_country;
+        const hasPhone = (await browser.storage.local.get("has_phone")).has_phone;
+        const hasVpn = (await browser.storage.local.get("has_vpn")).has_vpn;
+
+        // Conditions for phone masking announcement to be shown: if the user is in US/CAN, phone flag is on, and user has not purchased phone plan yet
+        const isPhoneMaskingAvailable = isPhoneAvailableInCountry && !hasPhone;
+
+        // Conditions for bundle announcement to be shown: if the user is in US/CAN, bundle flag is on, and user has not purchased bundle plan yet
+        const isBundleAvailable = isBundleAvailableInCountry && !hasVpn;
+      
+        // Conditions for firefox integration to be shown: if the waffle flag "firefox_integration" is set as true
+        const isFirefoxIntegrationAvailable = await checkWaffleFlag("firefox_integration");
+        
+        // FIXME: The order is not being set correctly
```

Comment:
Question: What if we added an `order` field and the number represents the order we want it in then we just sort the final array (new map) by that field?

Reply:
Yup! This was exactly @Vinnl's suggestion during the previous review of the news panel. On my TODO list.

Reply:
Resolved the order setting as part of this commit: cdd0de9c8fd3edec29206c4307c1100762298602

I added a field (`dateAdded`) to each object to give it some order, and then sort based on the date.